### PR TITLE
[Integate API] [UI] Create Homework and Delete Homework

### PR DESCRIPTION
**Pull request by timt1028**
resolve: #139 #138 

截圖 2021-06-28 上午12 04 33
截圖 2021-06-28 上午12 04 43

### CODE REVIEW

File: src/services/HomeworkService.js [hunk 1]
```diff
@@ -21,8 +21,9 @@ export class HomeworkService extends AbstractService {
             .then(res => res.data.map(obj => new Homework(obj)));
     }
 
-    async createHomework({name, problemIds}) {
-        return this.axios.post('/api/homework', new Homework({name, problemIds}))
+    async createHomework(name, problemList) {
+        const problemIds = problemList.split("\n");
```
Comment (Johnny850807) [inline src/services/HomeworkService.js:25]: problemIds should have been split as an argument

File: src/app/homework/modals/CreateHomeworkModal.js [hunk 2]
```diff
@@ -4,17 +4,25 @@ import {createRef, useState} from "react";
...
     const [name, setName] = useState(undefined);
     const [content, setContent] = useState();
+
     const nameInputRef = createRef();
     const closeIconRef = createRef();
     let formRef;
 
     const handleSubmit = e => {
-        e.preventDefault();
+        homeworkService.createHomework(name, content)
```
Comment (Johnny850807) [inline src/app/homework/modals/CreateHomeworkModal.js:20]: content is split by '\n' here

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: src/app/homework/modals/CreateHomeworkModal.js [hunk 3]
```diff
@@ -4,17 +4,25 @@ import {createRef, useState} from "react";
...
 import {ModalInput} from "../../commons/modals/ModalInput";
 import {ModalHeader} from "../../commons/modals/ModalHeader";
+import {homeworkService} from "../../../services/services";
 
 
-const CreateHomeworkModal = ({show, onClose}) => {
+const CreateHomeworkModal = ({show, onClose, onHomeworkCreated}) => {
+
     const [name, setName] = useState(undefined);
     const [content, setContent] = useState();
```
Comment (Wally5077) [inline src/app/homework/modals/CreateHomeworkModal.js:13]:
```Js
const [content, setContent] = useState(undefined);
```

File: src/app/homework/modals/CreateHomeworkModal.js [hunk 4]
```diff
@@ -4,17 +4,25 @@ import {createRef, useState} from "react";
...
     let formRef;
 
     const handleSubmit = e => {
-        e.preventDefault();
+        homeworkService.createHomework(name, content)
+            .then(homework => onHomeworkCreated(homework));
+        closeIconRef.current.click();
+        setName('');
+        setContent('');
+        e.preventDefault()
```
Comment (Wally5077) [inline src/app/homework/modals/CreateHomeworkModal.js:25]: e.preventDefault(); 放在第一行

File: src/app/homework/HomeworkList.js [hunk 5]
```diff
@@ -83,6 +88,7 @@ const HomeworkList = () => {
                                          ]}
                                          show={showDeleteHomeworkConfirmationModal}
                                          onClose={() => setShowDeleteHomeworkConfirmationModal(false)}
+                                         onSubmit={() => deleteHomework(selectedHomework.id)}
                 />
```
Comment (Wally5077) [inline src/app/homework/HomeworkList.js:92]: 放在上一行最後

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: src/app/homework/HomeworkList.js [hunk 6]
```diff
@@ -83,7 +88,8 @@ const HomeworkList = () => {
                                          ]}
                                          show={showDeleteHomeworkConfirmationModal}
                                          onClose={() => setShowDeleteHomeworkConfirmationModal(false)}
-                />
+                                         onSubmit={() => deleteHomework(selectedHomework.id)}/>
+
```
Comment (Wally5077) [inline src/app/homework/HomeworkList.js:93]: NIT: 這邊有多餘行數嗎？

File: src/app/homework/modals/CreateHomeworkModal.js [hunk 7]
```diff
@@ -4,17 +4,26 @@ import {createRef, useState} from "react";
...
-    const [content, setContent] = useState();
+    const [content, setContent] = useState(undefined);
+
     const nameInputRef = createRef();
     const closeIconRef = createRef();
     let formRef;
 
     const handleSubmit = e => {
-        e.preventDefault();
+        e.preventDefault()
```
Comment (Wally5077) [inline src/app/homework/modals/CreateHomeworkModal.js:20]: 分號

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: src/app/homework/HomeworkList.js [hunk 8]
```diff
@@ -26,13 +26,18 @@ const HomeworkList = () => {
     }, [homeworkList, refetchHomework]);
 
     const onHomeworkCreated = () => {
-        // TODO: here mapping issue " [UI] Create Homework Dialog And Delete Homework Dialog #139 "
+        refetchHomework();
     };
 
     if (!homeworkList) {
         return <Spinner/>
```
Comment (nyngwang) [inline src/app/homework/HomeworkList.js:33]: ;

File: src/app/homework/HomeworkList.js [hunk 9]
```diff
@@ -26,13 +26,18 @@ const HomeworkList = () => {
...
     }
 
+    const deleteHomework = (homeworkId) => {
+        homeworkService.deleteHomework(homeworkId)
+            .then(() => setHomeworkList(homeworkList.filter(homework => homework.id !== homeworkId)));
+    }
```
Comment (nyngwang) [inline src/app/homework/HomeworkList.js:39]: ;

File: src/app/homework/HomeworkList.js [hunk 10]
```diff
@@ -83,7 +88,8 @@ const HomeworkList = () => {
                                          ]}
                                          show={showDeleteHomeworkConfirmationModal}
                                          onClose={() => setShowDeleteHomeworkConfirmationModal(false)}
-                />
+                                         onSubmit={() => deleteHomework(selectedHomework.id)}/>
+
 
             </div>
         </div>
```
Comment (nyngwang) [inline src/app/homework/HomeworkList.js:95]: ;

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review Done.

**Review by Johnny850807 — APPROVED** [review]
LGTM

File: src/app/homework/HomeworkList.js (hunk 10 above)
Comment (timt1028) [reply src/app/homework/HomeworkList.js:95]: It's the html . Not use ':'

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by nyngwang — APPROVED** [review]
LGTM.

**Pull request merged**